src/components/genres-list/genres-list.jsx

```diff
@@ -0,0 +1,30 @@
+import React from "react";
+import PropTypes from "prop-types";
+import MovieProp from "../../types";
+
+const GenresList = (props) => {
+  const {movies, onGenreClick} = props;
+  const genres = [`All genres`];
+  movies.map((movie) => genres.push(movie.genre));
+  const uniqueGenres = [...new Set(genres)].slice(0, 9);
+
+  return (
+    <ul className="catalog__genres-list">
+      {uniqueGenres.map((genre, i) => (
+        <li key={`${i}-item`} className="catalog__genres-item" onClick={(evt) => {
+          evt.preventDefault();
+          onGenreClick(genre, movies);
+        }}>
+          <a href="#" className="catalog__genres-link">{genre}</a>
+        </li>
+      ))}
+    </ul>
+  );
+};
+
+GenresList.propTypes = {
+  movies: PropTypes.arrayOf(MovieProp),
```

Comment:
```suggestion
  movies: PropTypes.arrayOf(MovieProp).isRequired,
```
поскольку проверок в коде на то что movies могут быть не переданы нет и если разработчик забудет передать в компонент movies, приложение упадет еще на попытке сделать `movies.map` или `movies.forEach`